storage/pebble/registers.go

```diff
@@ -15,29 +16,39 @@ import (
 // Registers library that implements pebble storage for registers
 // given a pebble instance with root block and root height populated
 type Registers struct {
-	db           *pebble.DB
-	firstHeight  uint64
```

Comment:
I think we should keep this firstHeight, which is the first available height. The `FirstHeight()` method will read the latest value for `latestHeight` , and calculate the actual first height after taking the prune threshold and `firstHeight` into consideration.